packages/core-utils/src/logic-utils.ts

```diff
@@ -34,8 +34,8 @@ type SwitchCaseCase<T, U> = [T, () => U];
 type SwitchCaseDefault<U> = [typeof DEFAULT, () => U] | (() => U);
 
 type SwitchCase = {
-  <T, U>(value: T, ...args: [...SwitchCaseCase<T, U>[], SwitchCaseDefault<U>]): U;
-  <T, U>(value: T, ...args: SwitchCaseCase<T, U>[]): U | undefined;
+  <T extends string | number | boolean, U>(value: T, ...args: [...SwitchCaseCase<T, U>[], SwitchCaseDefault<U>]): U;
+  <T extends string | number | boolean, U>(value: T, ...args: SwitchCaseCase<T, U>[]): U | undefined;
```

Comment:
Based on the errors from the Terra UI build, it looks like this will also need to support `undefined`.

Reply:
Thank you for doing my debugging for me. I was chasing a red herring. Adding `undefined` seems to work. Should we add `null` now or wait for that to be added when it is needed?

Reply:
I would vote to go ahead and add `null`.